localstack/aws/connect.py

```diff
@@ -442,17 +457,77 @@ def get_client(
             Defaults to appropriate LocalStack endpoint.
         :param config: Boto config for advanced use.
         """
+        if config is None:
+            config = self._config
+        else:
+            config = self._config.merge(config)
+
+        endpoint_url = endpoint_url or get_local_service_url(service_name)
+        if service_name == "s3":
+            if re.match(r"https?://localhost(:[0-9]+)?", endpoint_url):
+                endpoint_url = endpoint_url.replace("://localhost", f"://{get_s3_hostname()}")
 
         return self._get_client(
             service_name=service_name,
             region_name=region_name or self._get_region(),
             use_ssl=self._use_ssl,
             verify=self._verify,
-            endpoint_url=endpoint_url or get_local_service_url(service_name),
-            aws_access_key_id=aws_access_key_id,
-            aws_secret_access_key=aws_secret_access_key,
+            endpoint_url=endpoint_url,
+            aws_access_key_id=aws_access_key_id or TEST_AWS_ACCESS_KEY_ID,
+            aws_secret_access_key=aws_secret_access_key or TEST_AWS_SECRET_ACCESS_KEY,
             aws_session_token=aws_session_token,
-            config=config or self._config,
+            config=config,
+        )
+
+
+class ExternalAwsClientFactory(ClientFactory):
+    def get_client(
+        self,
+        service_name: str,
+        region_name: Optional[str],
+        aws_access_key_id: Optional[str] = None,
+        aws_secret_access_key: Optional[str] = None,
+        aws_session_token: Optional[str] = None,
```

Comment:
question: we are not using those, right? As we explicitly set the client creation credentials to None to load them from our local configuration. Should we update the docs string to convey that?

Reply:
:+1: 

I'll sneak this into a later PR :wink:

Reply:
actually did this now, since I'm re-running the tests in -ext as well